thrifty-schema/src/test/kotlin/com/microsoft/thrifty/schema/LoaderTest.kt

```diff
@@ -992,6 +995,172 @@ class LoaderTest {
+        map.value.entries.size shouldBe 2
+        for ((key, value) in map.value.entries) {
+            assertTrue(key is LiteralValueElement)
+            val keyKey = (key as LiteralValueElement).value
+            when(keyKey) {
+                "locations" -> {
+                    val keyVal = (value as ListValueElement).value
+                    keyVal.size shouldBe 1
+                    val locationMap = keyVal[0] as MapValueElement
+                    locationMap.value.size shouldBe 3
+                    for ((locKey, locVal) in locationMap.value.entries) {
+                        when((locKey as LiteralValueElement).value) {
+                            "currencyCode" -> {
+                                assertTrue((locVal as LiteralValueElement).value == "USD")
+                            }
+                            "languages" -> {
+                                val languages = (locVal as ListValueElement).value
+                                languages.size shouldBe 2
+                                for (language in languages) {
+                                    val lang = (language as LiteralValueElement).value
+                                    assertTrue(lang in listOf("English", "Spanish"))
+                                }
+                            }
+                            "membership" -> {
+                                assertTrue((locVal as IdentifierValueElement).value == "Membership.MEMBER")
+                            }
+                            else -> {
+                                fail("Invalid const key; must be currencyCode, languages or membership")
+                            }
+                        }
+                    }
+                }
+                "isActive" -> {
+                    assertTrue((value as IdentifierValueElement).value == "true")
+                }
+                else -> {
+                    fail("Invalid const key; must be either locations or isActive")
+                }
+            }
+        }
+    }
+
+    @Test
+    fun invalidStructValuedConstant() {
+        val thrift = """
+            
+            struct Region {
+                1: required string name 
+                2: optional bool isActive
+            }
+
+            const UnknownRegion DEFAULT_REGION = {
+                "name": "US",
+                "nonActive" : true // field name does not exist in Region
+            }
+
+        """.trimIndent()
+
+        val e = shouldThrow<LoadFailedException> { load(thrift) }
+        e.message shouldContain "Failed to resolve type 'UnknownRegion'"
+    }
+
+    @Test
+    fun invalidKeyStructValuedConstant() {
+        val thrift = """
+            
+            struct Region {
+                1: required string name 
+                2: optional bool isActive
+            }
+
+            const Region DEFAULT_REGION = {
+                1 : "US",
+                "isActive" : true // field name does not exist in Region
```

Comment:
`isActive` does in fact exist; should the comment be for the line above this one?